src/kvirc/sparser/KviIrcServerParser_literalHandlers.cpp

```diff
@@ -954,9 +954,13 @@ void KviIrcServerParser::parseLiteralPrivmsg(KviIrcMessage * msg)
 		//			}
 		//		}
 
+		// "znc.in/self-message" capability: Handle a replayed message from ourselves to someone else.
+		bool bSelfMessage = IS_ME(msg, szNick);
+		QString szWindow = bSelfMessage ? szTarget : szNick;
```

Comment:
I'm not even using ZNC to test this.
`/inject :mynick!a@b PRIVMSG someoneelse!c@d :hi`